src/analysis/normed_space/weak_dual.lean

```diff
@@ -135,6 +135,59 @@ begin
   exact (inclusion_in_double_dual 𝕜 E z).continuous,
 end
 
+lemma to_weak_dual_image_eq (p : dual 𝕜 E → Prop) :
+  (to_weak_dual '' {x' : dual 𝕜 E | p x' }) =
+    {x' : weak_dual 𝕜 E | p x'.to_normed_dual } :=
+begin
+  ext x',
+  split,
+  { rintros ⟨x'', ⟨h₁, h₂⟩⟩, rw ← h₂, exact h₁, },
+  { intros h, use x', exact ⟨h, rfl⟩, },
+end
+
+lemma to_weak_dual_image_closed_unit_ball :
+  (to_weak_dual '' metric.closed_ball (0 : dual 𝕜 E) 1) =
+    {x' : weak_dual 𝕜 E | ∥ x'.to_normed_dual ∥ ≤ 1} :=
+begin
+  have eq : metric.closed_ball (0 : dual 𝕜 E) 1 = {x' : dual 𝕜 E | ∥ x' ∥ ≤ 1},
+  { ext1 x', simp only [dist_zero_right, metric.mem_closed_ball, set.mem_set_of_eq], },
+  rw eq,
+  exact to_weak_dual_image_eq _,
+end
+
 end normed_space.dual
 
 end weak_star_topology_for_duals_of_normed_spaces
+
+section polar_sets_in_weak_dual
+
+open metric set normed_space
+
+namespace polar
+
+variables {𝕜 : Type*} [nondiscrete_normed_field 𝕜]
+variables {E : Type*} [normed_group E] [normed_space 𝕜 E]
+
+/-- The polar `polar 𝕜 s` of a set `s : E` is a closed subset when the weak star topology
+is used, i.e., when `polar 𝕜 s` is interpreted as a subset of `weak_dual 𝕜 E`. -/
+lemma is_weak_dual_closed (s : set E) : is_closed (dual.to_weak_dual '' polar 𝕜 s) :=
```

Comment:
I wonder if you really want to use an explicit map from the dual to the weak dual, instead of just telling to the system to see this as a set in the weak dual. This could be done for instance with
`lemma is_weak_dual_closed (s : set E) : is_closed (id (polar 𝕜 s) : set (weak_dual 𝕜 E)) :=`
(where the `id` trick is to tell Lean that it should really listen to you).
Even if you want to use an explicit map, please avoid direct images (which are a pain to manipulate), and use instead preimages (i.e., formulate your theorem as `is_closed (weak_dual.to_dual ⁻¹'  (polar 𝕜 s))`). The reason is that saying that you are in a direct image means: there exists `x` in ... such that `y = f x`. While being in the preimage just means `f x \in ...` (so, no existential involved).

Reply:
Thank you!

I originally wrote polar sets directly in weak dual, and things were quite smooth, but the choice was clearly strange. When defining polar sets in dual, and talking about their properties in the weak-star topology, I didn't put enough thought into what "definition" is easiest to work with, I just took the image under the obvious bijection. Now that you say, it is clear that preimages under the inverse bijection are nicer to work with. I trust you that the `id` trick is good as well, and I initially did something like that (coercing to a set in the `weak_dual`), but I confused myself, because although Lean knew how to make sense of it, I no longer knew the definition...

For now I tested that at least preimages under `weak_dual.to_normed_dual` work and are an improvement. I will still at least try if I can understand how to work with the `id`.

Reply:
By the way, whatever the definition of the polar sets sets in the weak dual, perhaps this merits a definition or an abbreviation. Using preimages and an actual `def`, that would be
```
def weak_dual.polar (s : set E) : set (weak_dual 𝕜 E) := to_normed_dual ⁻¹' (polar 𝕜 s)
```

It makes at least many statements nicer. An `abbreviation` should work equally well, though.